### PR TITLE
Add components for data grid filters with selection inputs

### PR DESCRIPTION
**Pull request by ps37**
### Following are a list of filters with selection input:
Single select filter
Multi-select filter
Single select filter with custom FIQL

### Single select filter example:
selectFilter

### Multi-select filter example:
multiSelectFilter

### CODE REVIEW

File: projects/components/src/datagrid/filters/datagrid-multiselect-filter.component.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        return new FilterBuilder().any(...selectedFilters).getString();
+    }
+
+    isActive(): boolean {
+        return (
+            this.formGroup &&
+            !!Object.keys(this.formGroup.getRawValue()).filter(frmCtrl => this.formGroup.get(frmCtrl).value).length
+        );
+    }
+}
```
Comment (ryan-bradford) [inline projects/components/src/datagrid/filters/datagrid-multiselect-filter.component.ts]: Can you add a factory function to these filters like in `DatagridNumericFilter`? Also, how will this interact with the dynamic options?

File: projects/components/src/datagrid/filters/datagrid-select-filter.component.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    };
+
+    constructor(@Host() private filterContainer: ClrDatagridFilter, private fb: FormBuilder) {
+        super(filterContainer);
+        this.formGroup = this.fb.group({
+            filterSelect: '',
+        });
+    }
+
+    ngOnInit(): void {
```
Comment (juanmendes) [inline projects/components/src/datagrid/filters/datagrid-select-filter.component.ts]: Do we really need to debounce a select element?

File: projects/components/src/datagrid/filters/datagrid-multiselect-filter.component.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        if (this._config.value) {
+            this.setValue(this._config.value);
+        }
+    }
+    get config(): DatagridMultiSelectFilterConfig {
+        return this._config;
+    }
+
+    ngOnInit(): void {
+        this.debounceChanges(this.formGroup.valueChanges);
```
Comment (juanmendes) [inline projects/components/src/datagrid/filters/datagrid-multiselect-filter.component.ts]: Do we really need to debounce checkboxes? Typing makes sense, but most people will take at least 500ms between clicks

File: projects/components/src/utils/test/datagrid/filter-utils.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ import {
     GridDataFetchResult,
 } from '../../../datagrid';
 import { MockRecord } from '../../../datagrid/mock-data';
+import { MockTranslationService, TranslationService } from '@vcd/i18n';
```
Comment (juanmendes) [inline projects/components/src/utils/test/datagrid/filter-utils.ts]: Please have imports from external packages be at the top

File: projects/components/src/datagrid/filters/datagrid-multiselect-filter.component.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    getValue(): string {
+        const selectedFilters = Object.keys(this.formGroup.getRawValue())
+            .filter(frmCtrl => this.formGroup.get(frmCtrl).value)
+            .map(selectedOption => new FilterBuilder().is(this.queryField).equalTo(selectedOption));
+        return new FilterBuilder().any(...selectedFilters).getString();
+    }
+
+    isActive(): boolean {
+        return (
```
[the remark this answers is not in the document]
Comment (ps37) [reply projects/components/src/datagrid/filters/datagrid-multiselect-filter.component.ts]: That's Prettier adding the paranthesis

File: projects/components/src/utils/id-generator/id-generator.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+/**
+ * Object containing the logic to generate unique identifiers whenever it's {@link IdGenerator#id} method is called
```
Comment (juanmendes) [inline projects/components/src/utils/id-generator/id-generator.ts]: Object containing is redundant. Please avoid words like class, object, interface, logic, code, called in documentation.

Generates unique IDs.

File: projects/components/src/utils/id-generator/id-generator.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+ */
+export class IdGenerator {
+    /**
+     * Variable that is incremented whenever a new ID is needed
```
Comment (juanmendes) [inline projects/components/src/utils/id-generator/id-generator.ts]: No need for this comment. private variables should be documented if they are confusing.

File: projects/components/src/utils/id-generator/id-generator.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+     */
+    private static idCounter = 0;
+
+    /**
+     * Getter that returns the a unique ID
```
Comment (juanmendes) [inline projects/components/src/utils/id-generator/id-generator.ts]: No need to say getter, the code already says that. Having said that, a getter that always returns a different value seems wrong. Please rename to `generate` following the pattern from https://mongodb.github.io/mongo-java-driver/3.8/javadoc/org/bson/codecs/IdGenerator.html

**Pull request merged**